pootle/apps/accounts/views.py

```diff
@@ -7,16 +7,36 @@
 # or later license. See the LICENSE file for a copy of the license and the
 # AUTHORS file for copyright and authorship information.
 
+import logging
+
 from django.core.urlresolvers import reverse
 from django.shortcuts import redirect
 
 from allauth.account.views import LoginView
+from allauth.exceptions import ImmediateHttpResponse
 from allauth.socialaccount.helpers import _add_social_account
 from allauth.socialaccount.models import SocialLogin
 
 from .forms import SocialVerificationForm
 
 
+logger = logging.getLogger(__name__)
+
+
+class PootleLoginView(LoginView):
+
+    def form_valid(self, form):
+        success_url = self.get_success_url()
+        try:
+            return form.login(self.request, redirect_url=success_url)
+        except ImmediateHttpResponse as e:
+            return e.response
+        except Exception as e:
+            logger.error("%s %s" % (e.__class__.__name__, e))
+            raise RuntimeError("An error occurred logging you in. Please "
```

Comment:
I believe so - but i can change it to "an error occurred while logging you in" if thats clearer